dissect/target/plugins/os/windows/recyclebin.py

```diff
@@ -125,6 +129,44 @@ def read_bin_file(self, bin_path: TargetPath) -> RecycleBinRecord:
             _user=user,
         )
 
+    def read_recycle_deleted_folder(self, folder_path: TargetPath) -> RecycleBinRecord:
+        """
+        Generally speaking when deleting a file, the $R* file is the actual renamed deleted file.
+        This is however also the case for deleted folders. When a folder is deleted,
+        it is also renamed and placed here (with original recursive content).
+
+        This function will create RecycleBin records for each file in a deleted folder.
+        """
+
+        meta_file = self.target.fs.path(str(folder_path.parent / folder_path.name.replace("$R", "$I")).lstrip("/"))
+        if not meta_file.exists():
+            return
+
+        meta_data = meta_file.read_bytes()
+        header = self.select_header(meta_data)
+        entry = header(meta_data)
+
+        sid = self.find_sid(folder_path)
+        original_folder_path = self.target.fs.path(entry.filename.rstrip("\x00"))
+
+        user_details = self.target.user_details.find(sid=sid)
+        user = user_details.user if user_details else None
+        for parent_path, _, childs in folder_path.walk():
```

Comment:
Maybe `rglob` works better for you? Then you don't have to do the nasty path construction at line 156 and 157.